engine/src/state_chain_observer/client/subxt_state_chain_config.rs

```diff
@@ -0,0 +1,24 @@
+use subxt::{config::signed_extensions, Config};
+
+pub enum StateChainConfig {}
+
+impl Config for StateChainConfig {
+	type Hash = <state_chain_runtime::Runtime as frame_system::Config>::Hash;
+	type AccountId = subxt::utils::AccountId32;
+	type Address = subxt::utils::MultiAddress<Self::AccountId, ()>;
+	type Signature = subxt::utils::MultiSignature;
+	type Hasher = subxt::ext::sp_runtime::traits::BlakeTwo256;
+	type Header = subxt::ext::sp_runtime::generic::Header<u32, Self::Hasher>;
```

Comment:
```suggestion
	// We cannot use our own Runtime's types for every associated type here, see comments below.
	type Hash = <state_chain_runtime::Runtime as frame_system::Config>::Hash;
	type AccountId = subxt::utils::AccountId32; // Requires EncodeAsType trait (which our AccountId doesn't)
	type Address = subxt::utils::MultiAddress<Self::AccountId, ()>; // Must be convertible from Self::AccountId
	type Signature = state_chain_runtime::Signature
	type Hasher = subxt::ext::sp_runtime::traits::BlakeTwo256; // Requires subxt's custom Hash trait
	type Header = subxt::ext::sp_runtime::generic::Header<u32, Self::Hasher>; // Requires subxt's custom Header trait
```

Reply:
Yeah, I didn't think it was much better having them inconsistent. But not a strong opinion, will make the change.